pkg/kubectl/cmd/logs.go

```diff
@@ -162,18 +172,31 @@ func (o *LogsOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Com
 	o.ClientMapper = resource.ClientMapperFunc(f.ClientForMapping)
 	o.Out = out
 
+	if selector != "" {
+		if logOptions.Follow {
+			return cmdutil.UsageError(cmd, "only one of follow (-f) or selector (-l) is allowed")
```

Comment:
It has to do with interleaving logs (@smarterclayton was talking about) in case when getting logs from several pods.